File: tests/capabilities/typescript/positive.ts
// One deliberate trigger per capability rule in semgrep/capabilities/typescript.yml.
// Used by scripts/test-registration-conformance.py to confirm each rule still
// fires (recall). Not runnable application code.

import axios from "axios";
import * as child_process from "child_process";
import * as fs from "fs";
import jwt from "jsonwebtoken";
import nodemailer from "nodemailer";
import { S3Client } from "@aws-sdk/client-s3";
import { PrismaClient } from "@prisma/client";

function networkConnectionsExample() {
  return axios.get("https://example.invalid/api");
}

function persistenceExample() {
  return new PrismaClient();
}

function authenticationExample(payload: object) {
  return jwt.sign(payload, "secret");
}

function elevatedAccessExample() {
  return child_process.exec("echo hello");
}

function writeOrDeleteActionsExample(path: string) {
  fs.rmSync(path, { recursive: true, force: true });
}

function cloudExample() {
  return new S3Client({});
}

function externalRecipientsExample() {
  return nodemailer.createTransport({ host: "smtp.example.invalid" });
}

File: tests/capabilities/python/positive.py
"""One deliberate trigger per capability rule in semgrep/capabilities/python.yml.
Used by scripts/test-registration-conformance.py to confirm each rule still
fires (recall). Not runnable application code.
"""
import boto3
import bcrypt
import smtplib
import subprocess
import sqlite3
import requests


def network_connections_example():
    requests.get("https://example.invalid/api")


def persistence_example():
    conn = sqlite3.connect("app.db")
    return conn


def authentication_example(password: bytes):
    return bcrypt.hashpw(password, bcrypt.gensalt())


def elevated_access_example():
    subprocess.run(["echo", "hello"])


def write_or_delete_actions_example(path):
    import shutil
    shutil.rmtree(path)


def cloud_example():
    return boto3.client("s3")


def external_recipients_example():
    server = smtplib.SMTP("smtp.example.invalid")
    return server
